Improvement/1099 set kubeconfig in master config

TeddyAndrieux opened the pull request:
**Component**:

'salt', 'kubernetes'
<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

We need to have a `kubeconfig` to interact with kubernetes API, before this PR we give the `kubeconfig` to each and every call we do.

**Summary**:

Store the `kubeconfig` and `context` directly in the pillar and salt-master config

**Acceptance criteria**: 

Working bootstrap, upgrade, downgrade

---

Closes: #1099 
Closes: #1100

TeddyAndrieux committed:
salt: Add `kubeconfig` path in salt-master config

Fixes: #1099

TeddyAndrieux committed:
salt: Abilty to use `kubeconfig` from pillar or salt-master config

Salt module `metalk8s_kubernetes` need a kubeconfig to be able to talk
to APIServer. To simplify each and every call to this module add ability
to use a kubeconfig path stored in the pillar or in the salt-master
configuration file.

Fixes: #1099

TeddyAndrieux committed:
charts: Use new salt 'kubernetes' renderer for render

Use the refactored salt kubernetes renderer for charts render script to
no longer depend on the old kubernetes renderer.

Refs: #1048
(cherry picked from commit 979a5bbc29d506856bc65ae701eae552caa79124)

TeddyAndrieux committed:
script: Do not specify `kubeconfig` in upgrade/downgrade

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` in nginx-ingress chart

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
charts: Do not specify `kubeconfig` in charts renderer

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for prometheus-operator

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for calico

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for coredns

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for kube-proxy

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` to mark-control-plane

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for bootstrap orchestrate

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for deploy_node orchestrate

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for upgrade/downgrade orchestrate

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for repo deployment

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

TeddyAndrieux committed:
salt: Do not specify `kubeconfig` for salt-master deployment

We no longer need to specify `kubeconfig` and `context` when interacting
with API server as we have default value in pillar and salt master
config

Refs: #1099

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.4/improvement/1099-set-kubeconfig-in-master-config` with contents from `w/2.3/improvement/1099-set-kubeconfig-in-master-config`
and `development/2.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.4/improvement/1099-set-kubeconfig-in-master-config origin/development/2.4
 $ git merge origin/w/2.3/improvement/1099-set-kubeconfig-in-master-config
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/improvement/1099-set-kubeconfig-in-master-config
```

TeddyAndrieux committed:
script: Sync all mods in local for upgrade/downgrade

Sync all salt modules/states/... when doing a local call to upgrade or
downgrade the bootstrap node

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

NicolasT commented:
Do we really need to thread a `context` through? Can't we create a `KubeConfig` with the `current-context` set to the 'right' value (likely only one), and use that?

NicolasT reviewed (dismissed):
See question

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

TeddyAndrieux commented:
> Do we really need to thread a `context` through? Can't we create a `KubeConfig` with the `current-context` set to the 'right' value (likely only one), and use that?

Yes, we can basically just remove the context. `admin.conf` already have the right `current-context` set, even before this PR `context` was not necessary.

bert-e commented:
# History mismatch



Merge commit #108da880c30543529776b97d9eba65f648ae9c51 on the integration branch
`w/2.1/improvement/1099-set-kubeconfig-in-master-config` is merging a branch which is neither the current
branch `improvement/1099-set-kubeconfig-in-master-config` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1099-set-kubeconfig-in-master-config` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.4/improvement/1099-set-kubeconfig-in-master-config` with contents from `w/2.3/improvement/1099-set-kubeconfig-in-master-config`
and `development/2.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.4/improvement/1099-set-kubeconfig-in-master-config origin/development/2.4
 $ git merge origin/w/2.3/improvement/1099-set-kubeconfig-in-master-config
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/improvement/1099-set-kubeconfig-in-master-config
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.4/improvement/1099-set-kubeconfig-in-master-config` with contents from `w/2.3/improvement/1099-set-kubeconfig-in-master-config`
and `development/2.4`.

Please resolve the conflict on **the integration branch** (`w/2.4/improvement/1099-set-kubeconfig-in-master-config`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.4/improvement/1099-set-kubeconfig-in-master-config
 $ git pull  # or "git reset --hard origin/w/2.4/improvement/1099-set-kubeconfig-in-master-config"
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/2.3/improvement/1099-set-kubeconfig-in-master-config
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/improvement/1099-set-kubeconfig-in-master-config
```

gdemonet commented on salt/_modules/metalk8s_drain.py:
```diff
@@ -228,8 +228,7 @@ def get_controller(self, namespace, controller_ref):
             namespace=namespace,
-            kubeconfig=self._kwargs.get('kubeconfig'),
-            context=self._kwargs.get('context')
+            **self._kwargs
```
I thought you didn't want to do this because `self._kwargs` are the `Drain` object's kwargs? Minor though, feel free to leave them.

TeddyAndrieux replied:
But since we accept kwargs everywhere I think it's ok, at the beginning in kubernetes module we didn't have any kwargs I added it at the end (after your first comment so btw I should have done this before)

gdemonet commented on salt/_modules/metalk8s_kubernetes_utils.py:
```diff
@@ -40,7 +40,42 @@ def __virtual__():
+
+    kubeconfig = kwargs.get('kubeconfig') or \
+        pillar_dict.get('kubeconfig') or \
+        __salt__['config.option']('kubernetes.kubeconfig')
```
Don't you have access to `__opts__`? Or is that unrelated?

TeddyAndrieux replied:
We have access to `__opts__` just the initial `kubernetes` module from salt use this options so just kept it

gdemonet reviewed (dismissed):
Minor comments, LGTM

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1099-set-kubeconfig-in-master-config` into
`development/2.0`
* [w/2.1/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.1` 
* [w/2.2/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.2` 
* [w/2.3/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.3` 
* [w/2.4/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.4` 
* [w/2.5/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

bert-e commented:
# History mismatch



Merge commit #dca384929b1297a962e623b91c8b09b04bbb08d4 on the integration branch
`w/2.1/improvement/1099-set-kubeconfig-in-master-config` is merging a branch which is neither the current
branch `improvement/1099-set-kubeconfig-in-master-config` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/1099-set-kubeconfig-in-master-config` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.4/improvement/1099-set-kubeconfig-in-master-config` with contents from `w/2.3/improvement/1099-set-kubeconfig-in-master-config`
and `development/2.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.4/improvement/1099-set-kubeconfig-in-master-config origin/development/2.4
 $ git merge origin/w/2.3/improvement/1099-set-kubeconfig-in-master-config
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.4/improvement/1099-set-kubeconfig-in-master-config
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1099-set-kubeconfig-in-master-config` into
`development/2.0`
* [w/2.1/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.1` 
* [w/2.2/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.2` 
* [w/2.3/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.3` 
* [w/2.4/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.4` 
* [w/2.5/improvement/1099-set-kubeconfig-in-master-config](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/1099-set-kubeconfig-in-master-config) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @nicolast

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

TeddyAndrieux commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/74b58869aa5202cab0b683a72204311891a3048d) did not succeed in branch w/2.2/improvement/1099-set-kubeconfig-in-master-config.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye teddyandrieux.
